From 5879ab0f6999ed0672dc2deb860107a57eed0af5 Mon Sep 17 00:00:00 2001
From: MouchenHung <Mouchen.Hung@quantatw.com>
Date: Thu, 12 May 2022 14:18:24 +0800
Subject: [PATCH] MFZ_WC: First commit of new project

[Feature]
- Copy yv3.5 CraterLake platform code to new plarform code folder named
"mfz-wc"
- Modify "CMakeLists.txt" and "ast1030_evb.overlay" "sample.yaml"

[BugFix]
- none

[Note]
- CMakeLists.txt only modify I2C partition.
---
 meta-facebook/mfz-wc/CMakeLists.txt           |  2 +-
 .../mfz-wc/boards/ast1030_evb.overlay         | 36 ++++++-------------
 meta-facebook/mfz-wc/sample.yaml              |  2 +-
 .../mfz-wc/src/platform/plat_version.h        | 12 +++----
 4 files changed, 19 insertions(+), 33 deletions(-)

diff --git a/meta-facebook/mfz-wc/CMakeLists.txt b/meta-facebook/mfz-wc/CMakeLists.txt
index cab988b..a9f2cd2 100644
--- a/meta-facebook/mfz-wc/CMakeLists.txt
+++ b/meta-facebook/mfz-wc/CMakeLists.txt
@@ -3,7 +3,7 @@
 cmake_minimum_required(VERSION 3.13.1)
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
-project(yv35-cl)
+project(mfz-wc)
 
 set(common_path ../../common)
 FILE(GLOB app_sources src/ipmi/*.c src/platform/*.c src/lib/*.c)
diff --git a/meta-facebook/mfz-wc/boards/ast1030_evb.overlay b/meta-facebook/mfz-wc/boards/ast1030_evb.overlay
index 7ddc469..bcbb8d6 100644
--- a/meta-facebook/mfz-wc/boards/ast1030_evb.overlay
+++ b/meta-facebook/mfz-wc/boards/ast1030_evb.overlay
@@ -22,18 +22,18 @@
 &i2c0 {
 	pinctrl-0 = <&pinctrl_i2c0_default>;
 	status = "okay";
-  clock-frequency = <I2C_BITRATE_FAST>;
+	clock-frequency = <I2C_BITRATE_FAST>;
 };
 
 &i2c1 {
 	pinctrl-0 = <&pinctrl_i2c1_default>;
 	status = "okay";
-  clock-frequency = <I2C_BITRATE_STANDARD>;
+	clock-frequency = <I2C_BITRATE_STANDARD>;
 };
 
 &i2c2 {
 	pinctrl-0 = <&pinctrl_i2c2_default>;
-  clock-frequency = <I2C_BITRATE_FAST_PLUS>;
+	clock-frequency = <I2C_BITRATE_FAST_PLUS>;
 	status = "okay";
 
 	ipmb2: ipmb@10 {
@@ -51,24 +51,24 @@
 &i2c3 {
 	pinctrl-0 = <&pinctrl_i2c3_default>;
 	status = "okay";
-  clock-frequency = <I2C_BITRATE_STANDARD>;
+	clock-frequency = <I2C_BITRATE_STANDARD>;
 };
 
 &i2c4 {
 	pinctrl-0 = <&pinctrl_i2c4_default>;
 	status = "okay";
-  clock-frequency = <I2C_BITRATE_STANDARD>;
+	clock-frequency = <I2C_BITRATE_STANDARD>;
 };
 
 &i2c5 {
 	pinctrl-0 = <&pinctrl_i2c5_default>;
 	status = "okay";
-  clock-frequency = <I2C_BITRATE_FAST>;
+	clock-frequency = <I2C_BITRATE_FAST>;
 };
 
 &i2c6 {
 	pinctrl-0 = <&pinctrl_i2c6_default>;
-  clock-frequency = <I2C_BITRATE_FAST>;
+	clock-frequency = <I2C_BITRATE_FAST>;
 	status = "okay";
 
 	ipmb6: ipmb@20 {
@@ -82,24 +82,13 @@
 
 &i2c7 {
 	pinctrl-0 = <&pinctrl_i2c7_default>;
-  clock-frequency = <I2C_BITRATE_FAST_PLUS>;
 	status = "okay";
-
-	ipmb7: ipmb@20 {
-		compatible = "aspeed,ipmb";
-		reg = <0x20>;
-		label = "IPMB_7";
-		size = <10>;
-#ifdef CONFIG_I2C_IPMB_SLAVE
-		status = "okay";
-#endif
-	};
-
+	clock-frequency = <I2C_BITRATE_STANDARD>;
 };
 
 &i2c8 {
 	pinctrl-0 = <&pinctrl_i2c8_default>;
-  clock-frequency = <I2C_BITRATE_FAST_PLUS>;
+	clock-frequency = <I2C_BITRATE_FAST>;
 	status = "okay";
 
 	ipmb8: ipmb@20 {
@@ -107,17 +96,14 @@
 		reg = <0x20>;
 		label = "IPMB_8";
 		size = <10>;
-#ifdef CONFIG_I2C_IPMB_SLAVE
 		status = "okay";
-#endif
 	};
-
 };
 
 &i2c9 {
 	pinctrl-0 = <&pinctrl_i2c9_default>;
-	status = "disabled";
-
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
 };
 
 &espi {
diff --git a/meta-facebook/mfz-wc/sample.yaml b/meta-facebook/mfz-wc/sample.yaml
index d130e3f..6d691a1 100644
--- a/meta-facebook/mfz-wc/sample.yaml
+++ b/meta-facebook/mfz-wc/sample.yaml
@@ -2,7 +2,7 @@ tests:
   portability.cmsis_rtos_v2:
     tags: cmsis_rtos
 sample:
-  description: yv35_tmp 
+  description: mfz_tmp
   name: tmp
 common:
     tags: introduction
diff --git a/meta-facebook/mfz-wc/src/platform/plat_version.h b/meta-facebook/mfz-wc/src/platform/plat_version.h
index b9cc781..24cd726 100644
--- a/meta-facebook/mfz-wc/src/platform/plat_version.h
+++ b/meta-facebook/mfz-wc/src/platform/plat_version.h
@@ -1,13 +1,13 @@
 #ifndef PLAT_VERSION_H
 #define PLAT_VERSION_H
 
-#define PLATFORM_NAME "yv35"
-#define PROJECT_NAME "craterlake"
+#define PLATFORM_NAME "mfz"
+#define PROJECT_NAME "WaimeaCanyon"
 #define IANA_ID 0x009c9c
 #define DEVICE_ID 0x00
 #define DEVICE_REVISION 0x80
-#define FIRMWARE_REVISION_1 0x11
-#define FIRMWARE_REVISION_2 0x11
+#define FIRMWARE_REVISION_1 0x00
+#define FIRMWARE_REVISION_2 0x00
 #define IPMI_VERSION 0x02
 #define ADDITIONAL_DEVICE_SUPPORT 0xBF
 #define PRODUCT_ID 0x0000
@@ -15,8 +15,8 @@
 
 #define BIC_FW_YEAR_MSB 0x20
 #define BIC_FW_YEAR_LSB 0x22
-#define BIC_FW_WEEK 0x17
-#define BIC_FW_VER 0x01
+#define BIC_FW_WEEK 0x00
+#define BIC_FW_VER 0x00
 #define BIC_FW_platform_0 0x63 // char: c
 #define BIC_FW_platform_1 0x6c // char: l
 #define BIC_FW_platform_2 0x00 // char: '\0'
-- 
2.17.1

